### ISSUE TITLE
Rewrite this Plugin

### ISSUE THREAD

**Issue by Taki21** [Inactive-to-Reactive/SurvivalGamesV4#20]
@remotevase 
@ImagicalGamer 
Afraid that i have to rewrite this plugin (different branch) we will even use @remotevase 's 2 seperate file strategy!
lots of bugs have appeared and the code is kinda corrupted! Please stop me if u think this is weird LOL

**Comment by Taki21** [comment]
Whoops! Lel where is the Guidelines? Oh well who cares xD

**Comment by remotevase** [comment]
yeah, sounds good. I understand what you mean. on one of the releases that I posted, the sign issue was fixed, but many others remained

**Comment by remotevase** [comment]
and as we attempted to fix those other issues, the sign issue reappeared. then, we fixed that, and got more issues. then we fixed those issues and for the sign issue. it's a viscous cycle that can only be solved by a rewrite

**Comment by Heroata** [comment]
I'ts the best decision :D What is 2 seperate file strategy?

**Comment by remotevase** [comment]
like with loader and all that...

**Comment by remotevase** [comment]
and splitting into multiple parts

**Comment by Taki21** [comment]
To be started soon :)

**Comment by Taki21** [comment]
@ImagicalGamer We need ur decision though

**Comment by Taki21** [comment]
ur plugin breh.

**Comment by remotevase** [comment]
Already started, just need to commit

**Comment by remotevase** [comment]
not on master branch

**Comment by remotevase** [comment]
I started! I am doing it the way that many other plugins do it: Seperating into events, tasks, commands, and others. All joined using one file, which is called loader in this case!

**Comment by NikeboyHU** [comment]
Critical bug, and not running normally that plugin, and players couldn't join the match.

25.06 09:22:11 [Server] Server thread/CRITICAL "Could not pass event 'pocketmine\event\player\PlayerRespawnEvent' to 'SurvivalGamesV4 v4.1.7': Call to a member function addEffect() on null on SurvivalGamesV4\SurvivalGamesV4
25.06 09:22:11 [Server] Server thread/CRITICAL Error: "Call to a member function addEffect() on null" (EXCEPTION) in "/SGV4.1.7.phar/src/SurvivalGamesV4/SurvivalGamesV4" at line 133
25.06 09:22:19 [Server] INFO Warning: Illegal offset type in isset or empty in phar:///Genisys_1.2dev.phar/src/pocketmine/entity/Entity.php on line 381

And, that source code on php:


 	public function respawnWithEffects(PlayerRespawnEvent $e){
 		$p = $this->pl;
 		$s = Effect::getEffect(1);
 		$j = Effect::getEffect(8);
		$s->setDuration(2184728365782365723642365723652);
		$j->setDuration(2184728365782365723642365723652); 
		$s->setVisible(true);
		$j->setVisible(true);
		$s->setAmplifier(3);
		$j->setAmplifier(3);
		$p->addEffect($s);
		$p->addEffect($j);
 	}

Bug starting on line 133 ($p->addEffect($s); line), and hmmm looking this.

**Comment by remotevase** [comment]
Hey, guys! isn't it 
``` php
$p->giveEffect({effect});
```
not
``` php
$p->addEffect({effect});
```

**Comment by Taki21** [comment]
no ... i checked the docs...

**Comment by Taki21** [comment]
Null: Meaning that the function is meaningless so...there is a problem with $p = $this->pl;

**Comment by GlitchPlayer** [comment]
Signs not working... When will that fixed?

**Comment by remotevase** [comment]
OMG I ALMOST FORGOT ABOUT THIS PLUGIN! I will continue the rewrite later today and next week

**Comment by NikeboyHU** [comment]
Ohh yeah... 1 month later, than rewrite this, okey, good luck guys.
